src/rgw/setup/rgw.py

```diff
@@ -215,6 +218,18 @@ def _get_consul_url(conf: MappedConf, seq: int = 0):
         consul_url = 'consul:' + consul_fqdn + ':8500'
         return consul_url
 
+    @staticmethod
+    @ExponentialBackoff(exception=Exception, tries=4)
+    def _is_consul_alive(conf: MappedConf):
+        import socket
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        sock.settimeout(2)
+        consul_url = Rgw._get_consul_url(conf).split("//")[-1]
+        host, port = consul_url.split(':')
+        result = sock.connect_ex((host, port))
+        if result != 0:
+            raise SetupError(errno.EINVAL, "Consul server not alive")
```

Comment:
And message can be 
""Consul server not reachable through {ip:port}"

Reply:
done